backend/internal/observability/service.go

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// Package observability provides observability capabilities for Thunder including
+// event logging, metrics collection, distributed tracing, and health monitoring.
+package observability
+
+import (
+	"errors"
+	"path/filepath"
+	"sync"
+
+	"github.com/asgardeo/thunder/internal/observability/adapter"
+	"github.com/asgardeo/thunder/internal/observability/adapter/console"
+	"github.com/asgardeo/thunder/internal/observability/adapter/file"
+	"github.com/asgardeo/thunder/internal/observability/adapter/opensearch"
+	"github.com/asgardeo/thunder/internal/observability/event"
+	"github.com/asgardeo/thunder/internal/observability/formatter"
+	jsonformatter "github.com/asgardeo/thunder/internal/observability/formatter/json"
+	"github.com/asgardeo/thunder/internal/observability/publisher"
+	"github.com/asgardeo/thunder/internal/observability/subscriber/defaultsubscriber"
+	"github.com/asgardeo/thunder/internal/system/config"
+	"github.com/asgardeo/thunder/internal/system/log"
+)
+
+const loggerComponentName = "ObservabilityService"
+
+// Service provides observability event publishing functionality.
+// This is the main entry point for the observability system.
+// It manages the lifecycle of the publisher, which in turn manages subscribers.
+//
+// Architecture:
+//   - Service (High-level): Manages lifecycle, configuration, initialization
+//   - CategoryPublisher (Low-level): Implements event publishing logic
+//   - Subscribers (Low-level): Consume events
+//
+// The service layer owns the singleton pattern and initialization logic,
+// while the publisher layer provides the implementation.
+type Service struct {
+	publisher publisher.CategoryPublisher
+	logger    *log.Logger
+	config    *Config
+	enabled   bool
+}
+
+var (
+	serviceInstance *Service
+	serviceOnce     sync.Once
+)
+
+// GetService returns the singleton instance of the observability service.
+// Uses default configuration. For custom configuration, use InitializeWithConfig first.
+func GetService() *Service {
```

Comment:
What happens if this was called before calling InitializeWithConfig?

Reply:
We don't need to call the `InitializeWithConfig` before calling the getService. GetService will check if we need to load the config from the deployment.yaml or read by default